src/game/UICmdEdit.cpp

```diff
@@ -0,0 +1,76 @@
+// UICmdEditDlg.cpp: implementation of the CUICmdEditDlg class.
+//
+//////////////////////////////////////////////////////////////////////
+#include "StdAfx.h";
+#include "resource.h"
+#include "UICmdEdit.h"
+
+#include "GameProcMain.h"
+#include "GameProcedure.h"
+#include "PacketDef.h"
+#include "APISocket.h"
+
+//////////////////////////////////////////////////////////////////////
+// Construction/Destruction
+//////////////////////////////////////////////////////////////////////
+
+CUICmdEditDlg::CUICmdEditDlg() {}
+
+CUICmdEditDlg::~CUICmdEditDlg() {}
+
+bool CUICmdEditDlg::Load(HANDLE hFile) {
+    if (CN3UIBase::Load(hFile) == false) {
+        return false;
+    }
+
+    m_pText_Title = (CN3UIString *)GetChildByID("Text_cmd");
+    __ASSERT(m_pText_Title, "NULL UI Component!!");
+    m_pBtn_Ok = (CN3UIButton *)GetChildByID("btn_ok");
+    __ASSERT(m_pBtn_Ok, "NULL UI Component!!");
+    m_pBtn_Cancel = (CN3UIButton *)GetChildByID("btn_cancel");
+    __ASSERT(m_pBtn_Cancel, "NULL UI Component!!");
+    m_pEdit_Box = (CN3UIEdit *)GetChildByID("edit_cmd");
+    __ASSERT(m_pEdit_Box, "NULL UI Component!!");
+    return true;
+}
+
+bool CUICmdEditDlg::ReceiveMessage(CN3UIBase * pSender, DWORD dwMsg) {
+    if (dwMsg == UIMSG_BUTTON_CLICK) {
+        if (pSender->m_szID == "btn_ok") {
```

Comment:
We already have the pointer to this button to check against; can just check it directly instead, i.e.:
```cpp
if (pSender == m_pBtn_Ok)
```